src/Peachpie.Runtime/Reflection/TypeFields.cs

```diff
@@ -41,7 +41,7 @@ public class TypeFields
 
         internal TypeFields(Type tinfo)
         {
-            _fields = tinfo.GetFields(BindingFlags.Instance | BindingFlags.DeclaredOnly).Where(_IsAllowedField).ToDictionary(_FieldName, StringComparer.Ordinal);
+            _fields = tinfo.GetFields().Where(_IsAllowedField).ToDictionary(_FieldName, StringComparer.Ordinal);
```

Comment:
as I know, this has a different semantic - we MUST collect only declared fields (not fields from base type)